athina/evals/function/function_evaluator.py

```diff
@@ -0,0 +1,109 @@
+
+from typing import Optional
+
+from athina.metrics.metric_type import MetricType
+import time
+from typing import Optional
+from athina.interfaces.result import EvalResult, EvalResultMetric
+from athina.helpers.logger import logger
+from athina.interfaces.athina import AthinaExperiment
+from ..base_evaluator import BaseEvaluator
+from .functions import operations
+
+class FunctionEvaluator(BaseEvaluator):
+
+    _function_name: str
+    _function_arguments: dict
+    _experiment: Optional[AthinaExperiment] = None
+    _model: str
+
+    """
+    This evaluator runs the requested Function on the given data.
+    """
+
+    @property
+    def _model(self):
+        return ""
+    
+    @property
+    def name(self):
+        return self._function_name
+
+    @property
+    def display_name(self):
+        return "Function evaluator"
+
+    @property
+    def metric_ids(self) -> str:
+        return MetricType.PASSED.value
+
+    @property
+    def default_model(self):
+        return None
+
+    @property
+    def default_function(self):
+        return "contains_any"
+
+    @property
+    def default_function_arguments(self):
+        return {}
+
+    @property
+    def required_args(self):
+        return ["response"]
+
+    @property
+    def examples(self):
+        return None
+
+    def __init__(
+        self,
+        function_name: Optional[str] = None,
+        function_arguments: Optional[dict] = None,
+    ):
+        if function_name is None:
+            function_name = self.default_function
+        if function_arguments is None:
+            function_arguments = self.default_function_arguments
+        if function_name not in operations.keys():
+            raise ValueError(f"Unsupported function: {function_name}")
+        else:
+            self._function_name = function_name
+            self._function_arguments = function_arguments
+
+
+    def _evaluate(self, **kwargs) -> EvalResult:
+        """
+        Run the Function evaluator.
+        """
+        start_time = time.time()
```

Comment:
I know we use `time.time()` elsewhere as well, but I recently learned it's better to use `time.perf_counter()`.

https://superfastpython.com/time-time-vs-time-perf_counter/#What_is_timeperf_counter

Reply:
Fixed